lib/logflare/teams.ex

```diff
@@ -64,4 +68,40 @@ defmodule Logflare.Teams do
   @doc "Returns an `%Ecto.Changeset{}` for tracking team changes"
   @spec change_team(Team.t()) :: Ecto.Changeset.t()
   def change_team(%Team{} = team), do: Team.changeset(team, %{})
+
+  @doc "List all teams for user"
+  @spec list_for_user(User.t()) :: list(Team.t())
+  def list_for_user(%User{email: email, id: user_id}) do
+    team_user_query =
+      from t in Team,
+        join: tu in TeamUser,
+        on: tu.email == ^email
+
+    team_query =
+      from t in Team,
+        where: t.user_id == ^user_id,
+        union: ^team_user_query,
+        preload: [:user, :team_users]
+
```

Comment:
Can consolidate the query into:
```
from t in Team,
left_join: tu in assoc(t, :team_users)
where: tu.email == ^email or t.user_id == ^user_id
```
will be more performant than the union.

Also, should not preload by default, only conditionally. Each preload is an additional db query.

Also, using email might not be a good idea, not sure if the TeamUser `:token` is the same as the User `:token` when copied over, would be better to filter on it if possible.

Reply:
won't work because `user_id` will be used to join `team_user`. Since the user isn't the owner of all `team_user` entities it won't find all the teams this user is a member of

`team_user` and `user` do not have matching tokens, we would need to add a new field to `team_user` with the `user` token to match with it which would require way more code changes

Reply:
actually I will try to left join with email vs union 👁️

Reply:
ok on 2nd point.

for 1st point, can join on the user email instead of using the assoc macro.

Reply:
joining on user email won't work either because the join will then apply the where clauses and fail:
```elixir
team_query =
      from t in Team,
        join: tu in TeamUser,
        on: tu.email == ^email,
        where: t.user_id == ^user_id,
        preload: [:user, :team_users]
```

Reply:
<img width="429" alt="Screenshot 2023-02-02 at 11 26 25" src="https://user-images.githubusercontent.com/1697301/216312680-e7a70132-46fe-4d47-8475-2ae088bdf251.png">
still need to do a distinct and do the check on the email but this is closer 😄